### PR TITLE
category-name and subcategory-name

### PR DESCRIPTION
**Pull request by marcinbutora**

### CODE REVIEW

File: training-service/src/app/content/category/component/category-name/category-name.component.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  styleUrls: ['./category-name.component.scss']
+})
+export class CategoryNameComponent implements OnInit {
+
+  name: string | undefined;
+
+  constructor(private route: ActivatedRoute) { }
+
+  ngOnInit() {
+    this.route.params.subscribe(p => this.name = p['name']);
```
Comment (holistic-fox) [inline training-service/src/app/content/category/component/category-name/category-name.component.ts]: Unsubscribe is missing for this subscription, this is a potential memory leak.

File: training-service/src/app/content/subcategory/component/subcategory-name/subcategory-name.component.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  styleUrls: ['./subcategory-name.component.scss']
+})
+export class SubcategoryNameComponent implements OnInit {
+
+  name: string | undefined;
+
+  constructor(private route: ActivatedRoute) { }
+
+  ngOnInit() {
+    this.route.params.subscribe(p => this.name = p['name'])
```
Comment (holistic-fox) [inline training-service/src/app/content/subcategory/component/subcategory-name/subcategory-name.component.ts]: Unsubscribe is missing for this subscription, this is a potential memory leak.

File: training-service/src/app/content/category/component/category-name/category-name.component.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-category-name',
+  templateUrl: './category-name.component.html',
+  styleUrls: ['./category-name.component.scss']
+})
+export class CategoryNameComponent implements OnInit {
```
Comment (holistic-fox) [inline training-service/src/app/content/category/component/category-name/category-name.component.ts]: CategoryComponent is a better name than CategoryNameComponent

**Review by holistic-fox — CHANGES_REQUESTED** [review]

File: training-service/src/app/content/category/component/category-name/category-name.component.ts (hunk 1 above)
Comment (marcinbutora) [reply training-service/src/app/content/category/component/category-name/category-name.component.ts]: done

File: training-service/src/app/content/subcategory/component/subcategory-name/subcategory-name.component.ts (hunk 2 above)
Comment (marcinbutora) [reply training-service/src/app/content/subcategory/component/subcategory-name/subcategory-name.component.ts]: done

File: training-service/src/app/content/category/component/category-name/category.component.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+@Component({
+  selector: 'app-category',
+  templateUrl: './category-name.component.html',
+  styleUrls: ['./category-name.component.scss']
+})
+export class CategoryNameComponent implements OnInit {
+
+  name: string | undefined;
+  paramSubscription: any;
```
Comment (holistic-fox) [inline training-service/src/app/content/category/component/category-name/category.component.ts]: Please use strict typing such as:
Subscription | undefined;

**Review by holistic-fox — CHANGES_REQUESTED** [review]

File: training-service/src/app/content/category/component/category-name/category.component.ts (hunk 4 above)
Comment (marcinbutora) [reply training-service/src/app/content/category/component/category-name/category.component.ts]: done

**Review by holistic-fox — APPROVED** [review]

**Pull request merged**